apps/dashboard/test/controllers/custom_pages_controller_test.rb

```diff
@@ -0,0 +1,28 @@
+require "test_helper"
+
+class CustomPagesControllerTest < ActiveSupport::TestCase
+
+  test "index should set flash message if page_code configuration is not found" do
+    target = CustomPagesController.new
+    target.instance_variable_set(:@user_configuration, stub("user_configuration", {custom_pages: {}}))
```

Comment:
can these be system or integration tests instead? (note system tests require some setup with chrome driver).

I feel like when there are so many stubs it's like yea, it's going to behave that way. And in any case, we don't really have many controller tests, so I'd rather buy into to either integration (checking the html of the page) or system (also checking the html but from a web browser).

Reply:
Creating integration test and checking HTML...